compiler/ast/ast.go

```diff
@@ -131,38 +154,59 @@ func (*QuotedString) PatternAST() {}
 func (*Regexp) PatternAST()       {}
 func (*String) PatternAST()       {}
 
+func (x *QuotedString) Pos() int { return x.Lquote }
+func (x *String) Pos() int       { return x.TextPos }
+
+func (x *QuotedString) End() int { return x.Lquote + 2 + len(x.Text) }
+func (x *String) End() int       { return x.TextPos + len(x.Text) }
```

Comment:
In the interest of keeping things consistent with our style elsewhere, are you up for
(a) moving the Pos and End methods so they come right after the receiver type declarations and 
(b) changing the receiver names from `x` to the first character of the type name?

Reply:
If you think this will help readability, sure.

Reply:
What do we want to do with ops considering that they are declared in a big `var ()` statement?

Reply:
I'm for breaking them out into a separate statement per type but let's do that later. If you agree, then I'd say put the methods for those types where ever you like.

Reply:
Alright, then we'll just leave the op stuff where it is and reformat this in a future pr once this gets in.